packages/chain-adapters/src/cosmossdk/osmosis/OsmosisChainAdapter.ts

```diff
@@ -32,30 +45,32 @@ export class ChainAdapter
       assetNamespace: 'slip44',
       assetReference: ASSET_REFERENCE.Osmosis
     })
-  }
 
-  getType(): ChainTypes.Osmosis {
-    return ChainTypes.Osmosis
+    this.parser = new unchained.cosmos.TransactionParser({ chainId: this.chainId })
   }
 
   getFeeAssetId(): AssetId {
     return 'cosmos:osmosis-1/slip44:118'
```

Comment:
Not osmosis chain adapter specific, but ideally this would more accurately be `this.assetId` which would work if we did have testnet support as well.

Reply:
cool. not gonna do that in this pr if its cool